optuna/study/study.py

```diff
@@ -535,8 +529,6 @@ def ask(
 
         # Sync storage once every trial.
         self._thread_local.cached_all_trials = None
-        if isinstance(self._storage, _CachedStorage):
-            self._storage.read_trials_from_remote_storage(self._study_id)
```

Comment:
[Notes] `_pop_waiting_trial_id` internally calls `get_all_trials`. So we can remove this line.
https://github.com/optuna/optuna/blob/f4eec0eea71a8155f1c32f14b7e20f95c061d97b/optuna/study/study.py#L1036-L1038